### PR TITLE
Microsoft Teams - enhancements 

### PR DESCRIPTION
**Pull request by Itay4**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/28350

## Minimum version of Demisto
- [x] 5.0.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 1]
```diff
@@ -766,88 +771,503 @@ def add_user_to_channel_command():
...
+    if not parts_to_clone:
+        raise ValueError('At least one of the parts of the team must be cloned: apps, tabs, settings, channels')
+    clone_team_request(
+        team_id=team_id,
+        display_name=demisto.args().get('display_name'),
+        description=demisto.args().get('description'),
+        visibility=demisto.args().get('visibility'),
+        parts_to_clone=','.join(parts_to_clone),
+    )
+    demisto.results(f'Team {team_id} was cloned successfully.')
```
Comment (yaakovi) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:1247]: same for all
```suggestion
    return_results(f'Team {team_id} was cloned successfully.')
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 2]
```diff
@@ -766,88 +771,503 @@ def add_user_to_channel_command():
...
+            'mailNickname': display_name,
+            'description': description,
+            'visibility': visibility,
+            'partsToClone': parts_to_clone,
+        }
+    )
+
+
+def clone_team():
+    team_id = demisto.args().get('team_id')
```
Comment (yaakovi) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:1228]: don't call `demisto.args()` every time, save that value as a variable.
same for all

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 3]
```diff
@@ -766,88 +771,503 @@ def add_user_to_channel_command():
...
+) -> None:
+    url = f'{GRAPH_BASE_URL}/v1.0/teams/{team_id}/clone'
+    http_request(
+        'POST',
+        url,
+        json_={
+            'displayName': display_name,
+            # from Microsoft docs: "mailNickname property must be specified when a group is created.
+            # If this property is not specified, it will be computed from the displayName.
+            # Known issue: this property is currently ignored."
```
Comment (yaakovi) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:1218]: nice

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 4]
```diff
@@ -766,88 +771,503 @@ def add_user_to_channel_command():
...
+            team_member,
+            ['id', 'displayName', 'email', 'roles'],
+        ),
+        outputs=team_member,
+        raw_response=team_member,
+    ))
+
+
+@logger
+def remove_member__request(team_id: str, membership_id: str) -> None:
```
Comment (yaakovi) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:1132]: i'm guessing it's by mistake
```suggestion
def remove_member_request(team_id: str, membership_id: str) -> None:
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 5]
```diff
@@ -766,88 +771,503 @@ def add_user_to_channel_command():
...
+
+
+def create_team_from_group():
+    group_id = demisto.args().get('group_id', '')
+    create_team_from_group_request(
+        group_id=group_id,
+        display_name=demisto.args().get('display_name'),
+        description=demisto.args().get('description'),
+        visibility=demisto.args().get('visibility', 'public'),
+        allow_guests_create_channels=argToBoolean(demisto.args().get('allow_guests_create_channels', 'false')),
```
Comment (yaakovi) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:913]: can we switch to the same format as in `update_team`?

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 6]
```diff
@@ -766,88 +771,503 @@ def add_user_to_channel_command():
...
+                'allowUserDeleteMessages': allow_user_delete_messages,
+                'allowOwnerDeleteMessages': allow_owner_delete_messages,
+                'allowTeamMentions': allow_team_mentions,
+                'allowChannelMentions': allow_channel_mentions,
+            },
+        },
+    )
+
+
+def create_team():
```
Comment (yaakovi) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:830]: same here

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams_test.py [hunk 7]
```diff
@@ -1419,3 +1386,714 @@ def test_integration_health(mocker):
...
+        - Verify human readable output
+    """
+    from MicrosoftTeams import create_team_from_group
+    group_id = 'uuid'
+    requests_mock.put(f'https://graph.microsoft.com/v1.0/groups/{group_id}/team', status_code=202)
+    mocker.patch.object(
+        demisto,
+        'args',
+        return_value={
+            'group_id': group_id,
```
Comment (yaakovi) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams_test.py:1471]: set one of the `allow_*` values too

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.yml [hunk 8]
```diff
@@ -291,7 +291,946 @@ script:
...
+      required: true
+      secret: false
+    deprecated: false
+    description: Creates a new team.
+    execution: false
+    name: microsoft-teams-create-team
+  - arguments:
+    - default: false
+      description: ID of group to create team from. Can be retrieved by running the
+        msgraph-groups-list-groups command.
```
Comment (yaakovi) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.yml:474]: nice

**Review by yaakovi — COMMENTED** [review]
looks very good 🥇 

why don't we add context in most of the commands?

**Comment by Itay4** [comment]
> looks very good 🥇
> 
> why don't we add context in most of the commands?

some of the APIs dont return content, just successful response status code
for example, [Create Team API](https://docs.microsoft.com/en-us/graph/api/team-post?view=graph-rest-1.0&tabs=http)

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py (hunk 5 above)
Comment (Itay4) [reply Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:913]: the difference between create-team commands and update command is that in the update command, i dont want to the bool args to have default value, as they might cause the user to update properties which were not meant to be updated - if that wasnt the case, i would implement the update command the same as the create commands
i can modify the create commands to be implemented in the same style as the update command is implemented, but i think the way it is more straight forward and simple
what do you think?

**Comment by ShirleyDenkberg** [comment]
@yaakovi Doc review completed.

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py (hunk 5 above)
Comment (yaakovi) [reply Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py]: it's fine as is

**Review by yaakovi — APPROVED** [review]
looks good to me, let me know when to review again

**Comment by Itay4** [comment]
@yaakovi ready for another review

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeamsManagement/MicrosoftTeamsManagement.yml [hunk 9]
```diff
@@ -0,0 +1,1037 @@
...
+      description: Description of the team.
+      isArray: false
+      name: description
+      required: false
+      secret: false
+    - auto: PREDEFINED
+      default: false
+      defaultValue: public
+      description: 'The visibility of the group and team. Possible values are: "public"
+        and "private". Default is "public".'
```
Comment (yaakovi) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeamsManagement/MicrosoftTeamsManagement.yml:62]: please remove all `Default is "*".` where `defaultValue` is specified.
it looks bad in the product (example from current Teams integration):
image

**Review by yaakovi — APPROVED** [review]

**Pull request merged**